src/ir/effects.h

```diff
@@ -454,7 +433,16 @@ struct EffectAnalyzer
         case DivUInt64:
         case RemSInt64:
         case RemUInt64: {
+          // div and rem may contain implicit trap only if RHS is
+          // non-constant or constant which equal zero or -1 for
+          // signed operations
           implicitTrap = true;
+          if (auto* c = curr->right->dynCast<Const>()) {
+            implicitTrap = c->value.isZero() ||
+                           ((curr->op == DivSInt32 || curr->op == DivSInt64 ||
+                             curr->op == RemSInt32 || curr->op == RemSInt64) &&
+                            c->value.getInteger() == -1LL);
```

Comment:
Does INT_MIN / -1 trap?

Reply:
yes. It's trapping with "integer overflow error"

Reply:
Realized `INT_MIN % -1` can't trap in wasm so removed check for reminders